openapi-types incompatible with TypeScript 4.8

Error:
```
node_modules/@fastify/swagger/node_modules/openapi-types/dist/index.d.ts:16:28 - error TS2344: Type 'T' does not satisfy the constraint '{}'.

16         paths: PathsObject<T>;
                              ~

  node_modules/@fastify/swagger/node_modules/openapi-types/dist/index.d.ts:15:34
    15     type PathsWebhooksComponents<T> = {
                                        ~
    This type parameter might need an `extends {}` constraint.
```

--- a/packages/openapi-types/index.ts
+++ b/packages/openapi-types/index.ts
@@ -36,7 +36,7 @@ export namespace OpenAPI {
 export namespace OpenAPIV3_1 {
   type Modify<T, R> = Omit<T, keyof R> & R;
 
-  type PathsWebhooksComponents<T> = {
+  type PathsWebhooksComponents<T extends {} = {}> = {
     paths: PathsObject<T>;
     webhooks: Record<string, PathItemObject | ReferenceObject>;
     components: ComponentsObject;

